### PR TITLE
Feature/error handling

### PR DESCRIPTION
**Pull request by itsnameissteven**
## Is this a fix or a feature?
Feature
## What is the change?
Adds error handling to api calls on registration form 
add test for error handling
refactors tests
## What does it fix?
users are redirected to appropriate page if something needs to be fixed
## Where should the reviewer start?
review the registration page and api calls

### CODE REVIEW

File: src/apiCalls.ts [hunk 1]
```diff
@@ -43,3 +43,12 @@ export const getUserByID = (id: number) => {
...
 }
+
+
+const handleErrors = (response: Response) => {
+  console.log(response)
+  if(!response.ok) {
+    throw new Error(`${response.status}`)
+  }
+  return response.json()
+}
```
Comment (alexthompson207) [inline src/apiCalls.ts:54]: Nice error handling here, I like this!

File: src/components/Error/Error.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { BsExclamation } from 'react-icons/bs'
+
+interface ErrorProps {
+  message: string
+}
+
+const Error: FC<ErrorProps> = ({message}) => {
+  return (
+    <p className="error">
+        <BsExclamation className="error__exclamation"/>{message}
```
Comment (alexthompson207) [inline src/components/Error/Error.tsx:11]: This is another clean error handling component, nice work!

File: src/components/Registration/Registration.tsx [hunk 3]
```diff
@@ -18,40 +20,60 @@ const Registration: FC<RegistrationProps> = (props) => {
...
+    } else if (page === 4) {
       postRouteData(routeData)
-        .then(() => setIsLoggedIn(true))
+        .then(() => {
+          setError(false);
+          setIsLoggedIn(true);
+        })
+        .catch(() => {
+          setError(true);
+          setPage(3);
```
Comment (alexthompson207) [inline src/components/Registration/Registration.tsx:72]: Makes sense to redirect the user back a page, this is great!

**Review by alexthompson207 — APPROVED** [review]

File: cypress/integration/hitch-error-handling_spec.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      method: "GET",
+      statusCode: 400
+    })
+    cy.intercept('https://afternoon-journey-49986.herokuapp.com/api/v1/users/1/rides', {
+      method: "GET",
+      statusCode: 400,
+      body: {}
+    })
+    cy.visit('http://localhost:3000/register');
+    cy.intercept("https://maps.googleapis.com/maps/api/place/js/*")
```
Comment (PaigeVannelli) [inline cypress/integration/hitch-error-handling_spec.js:94]: Interesting intercept!

File: src/apiCalls.ts [hunk 5]
```diff
@@ -43,3 +43,12 @@ export const getUserByID = (id: number) => {
...
     .then(response => response.json())
 }
+
+
+const handleErrors = (response: Response) => {
+  console.log(response)
+  if(!response.ok) {
+    throw new Error(`${response.status}`)
+  }
+  return response.json()
```
Comment (PaigeVannelli) [inline src/apiCalls.ts:53]: Looks good!

**Review by PaigeVannelli — APPROVED** [review]

**Pull request merged**